### DISCUSSION TITLE
Role of `private/` folders

### DISCUSSION THREAD

**Discussion by stestoll** [mathworks/toolboxdesign#13] [General]
Thanks for these Best Practices!

You are recommending an `internal` folder for code that's not intended for the end user. In my toolbox, I am using a `private` subfolder for that. What are your thoughts regarding this?

**Comment by rpurser47** [comment]
Hi Stefan, thanks for your thoughts on this!  There's nothing wrong with ``private`` folders.  MATLAB code inside the folder with the "magic name" can only be called from the directory above ([more info](https://www.mathworks.com/help/matlab/matlab_prog/private-functions.html)).  For small toolboxes, this is rarely a problem.  However, with larger toolboxes, people are sometimes surprised that they can't call one of their utility functions from another directory.  Thus, we recommend ``internal``.  I also have insight into where MATLAB is headed, and know that ``internal`` will work better in the future.
